pkg/migration-tool/reader/reader.go

```diff
@@ -93,5 +117,133 @@ func (rr *RemoteRead) Run(errChan chan<- error) {
 			rr.sigBlockRead <- blockRef
 			blockRef = nil
 		}
-	}()
+	}
+
+	readPromBlocks := func() {
+		// Comment.
+		defer deferFunc()
+		db, err := tsdb.OpenDBReadOnly(rr.path, nil)
+		if err != nil {
+			errChan <- fmt.Errorf("remote-read run: opening prom tsdb: %w", err)
+			return
+		}
+		defer func() {
+			err = tsdb_errors.NewMulti(err, db.Close()).Err()
+		}()
+		querier, err := db.Querier(rr.c, rr.plan.Mint(), rr.plan.Maxt())
+		if err != nil {
+			errChan <- fmt.Errorf("remote-read run: prom tsdb querier: %w", err)
+			return
+		}
+		defer querier.Close()
+		ss := querier.Select(false, nil, labels.MustNewMatcher(labels.MatchRegexp, "", ".*"))
+		var iteratorStore = new(sync.Map)
```

Comment:
why is this a sync.Map and not a regular map?

Reply:
No specific reason. I should convert to a regular map. I think, while implementing i though of a concurrent safe map, but i dont think its a requirement now.